deltachat-ios/AppDelegate.swift

```diff
@@ -481,12 +484,9 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationCenterD
 
                 // to avoid 0xdead10cc exceptions, scheduled jobs need to be done before we get suspended;
                 // we increase the probabilty that this happens by waiting a moment before calling completionHandler()
-                DispatchQueue.global().async { [weak self] in
-                    guard let self = self else { completionHandler(.failed); return }
-
-                    _ = fetchSemaphore.wait(timeout: .now() + 1)
+                DispatchQueue.global().async {
+                    usleep(1_000_000)
```

Comment:
have you read that this is better than asyncAfter or it more trial and error?

Reply:
i figured that out by testing: `DispatchQueue.main.asyncAfter` is very often executed only when the app back in foreground, so after tens of minutes. same for `DispatchQueue.global(qos: .background).async`.

best guarantee i found that things are really executed in time from a backround thread is a non-mainthread with higher qos, i tested `.default`, therefore `DispatchQueue.global().async`.

note: that all is true if the app is in background only, in foreground all variants are just fine.

Reply:
> best guarantee i found that things are really executed in time from a backround thread is a non-mainthread with higher qos, i tested `.default`, therefore `DispatchQueue.global().async`.

oh I ment `DispatchQueue.global().asyncAfter` not `DispatchQueue.main.asyncAfter`

Reply:
by the removal of two thread dispatching (by allowing to call appIsInForeground() from background), this question is no longer relevant :)